ext/zpages/test/tracez_processor_benchmark.cc

```diff
@@ -0,0 +1,191 @@
+#include "opentelemetry/context/threadlocal_context.h"
+#include "opentelemetry/ext/zpages/tracez_processor.h"
+
+#include <benchmark/benchmark.h>
+#include <thread>
+
+#include "opentelemetry/sdk/trace/tracer.h"
+
+using namespace opentelemetry::sdk::trace;
+using namespace opentelemetry::ext::zpages;
+
+/////////////////////////////// BENCHMARK HELPER FUNCTIONS //////////////////////////////
+
+/*
+ * Helper function that creates and add i spans into the passed in vector.
+ */
+void StartManySpans(
+    std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>> &spans,
+    std::shared_ptr<opentelemetry::trace::Tracer> &tracer,
+    int i)
+{
+  for (; i > 0; i--)
+    spans.push_back(tracer->StartSpan(""));
+}
+
+/*
+ * Helper function that ends and removes all spans in the passed in span vector,
+ * since they'll be in the processor memory.
+ */
+void EndAllSpans(std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>> &spans)
+{
+  while(!spans.empty())
+  {
+    auto span = spans.back();
+    span->End();
+    spans.pop_back();
+  }
+}
+
+/*
+ * Helper function calls GetSpanSnapshot() i times, does nothing otherwise.
+ * Snapshots are significant and contribute to performance differences because
+ * completed spans are cleared from the processor memory. This function
+ * simulates an aggregator querying the processor many times, but doesn't
+ * aggregate or store spans.
+ */
+void GetManySnapshots(std::shared_ptr<TracezSpanProcessor> &processor, int i)
+{
+  for (; i > 0; i--)
+    processor->GetSpanSnapshot();
+}
+
+////////////////////////  FIXTURE FOR SHARED SETUP CODE ///////////////////
+
+class TracezProcessor : public benchmark::Fixture
+{
+protected:
+  void SetUp(const ::benchmark::State& state)
+  {
+    processor  = std::shared_ptr<TracezSpanProcessor>(new TracezSpanProcessor());
+    tracer     = std::shared_ptr<opentelemetry::trace::Tracer>(new Tracer(processor));
+  }
+
+  std::shared_ptr<TracezSpanProcessor> processor;
+  std::shared_ptr<opentelemetry::trace::Tracer> tracer;
+
+  std::vector<opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span>> spans;
+};
+
+//////////////////////////// BENCHMARK DEFINITIONS /////////////////////////////////
+
+/*
+ * Make many empty spans. This checks the scenario where the processor holds
+ * many running spans but never gets queried.
+ */
+BENCHMARK_DEFINE_F(TracezProcessor, BM_Run)(benchmark::State &state)
```

Comment:
Nit: Update the benchmark names to more detailed, so it will be easy to understand what the benchmark is measuring without reading the comment (for example for someone that is looking at the benchmark's result). 
For example here you can name it as BM_ProcessStartSpans

Reply:
will do!!